File: ATCD.Backend.Dto/Web/SongSearchDto.cs
namespace ATCD.Backend.Dto.Web
{
    public class SongSearchDto
    {
        public string SearchText { get; set; }

        public int Page { get; set; }

        public int ItemsPerPage { get; set; }

        public List<byte> ChoreoExcludes { get; set; }

        public List<int> Genres { get; set; }

        public List<byte> ChoreoTypes { get; set; }

        public DateTime? PublishedFrom { get; set; }

        public DateTime? PublishedTo { get; set; }
    }
}

File: frontend/src/models/SongSearchDto.ts
export class SongSearchDto {
  searchText: string = ''
  page: number = 1
  itemsPerPage: number = 50
  choreoExcludes: number[] = []
  genres: number[] = []
  choreoTypes: number[] = []
  publishedFrom: Date | null | undefined
  publishedTo: Date | null | undefined
}
